src/SegmentedString.chpl

```diff
@@ -1429,11 +1438,8 @@ module SegmentedString {
     The SegFunction called by computeOnSegments for bytesToUintArr
   */
   inline proc stringBytesToUintArr(values, rng) throws {
-      var concat: uint;
-      for v in values[rng] {
-        concat = concat<<8 + v:uint;
-      }
-      return concat;
+      var localSlice = new lowLevelLocalizingSlice(values, rng);
```

Comment:
I don't feel like the localizing slice should be necessary since we're calling the using `computeOnSegments`, so things should be local? But I know a lot of the other computeOnSegments functions call either `interpretAsString` or `interpretAsBytes` which do a localizing slice, so it seemed worth a try